pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4208 @@
+            Name of the namespace containing the class that will be modified
+            (case independent).
+
+            Leading and trailing slash characters will be stripped. The lexical
+            case will be preserved.
+
+          ModifiedClass (:class:`~pywbem.CIMClass`):
+            A representation of the modified class.
+
+            The properties, methods and qualifiers defined in this object
+            specify what is to be modified.
+
+        Raises:
+
+            CIMError: CIM_ERR_NOT_SUPPORTED
+        """
+
+        raise CIMError(
+            CIM_ERR_NOT_SUPPORTED,
+            "Currently ModifyClass not supported in MainProvider")
+
+    ####################################################################
+    #
+    #   DeleteClass server request
+    #
+    ####################################################################
+
+    def DeleteClass(self, namespace, ClassName):
+        """
+        Implements a WBEM server responder for
+        :meth:`~pywbem.WBEMConnection.DeleteClass`
+
+        Delete a class in the class CIM repository if it exists.
+
+        This method also deletes any subclasses and CIM instances of
+        the target class or its subclasses.
+
+        Nothing is returned.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            Name of the namespace containing the class defined by ClassName that
+            will be deleted (case independent).
+
+            Leading and trailing slash characters will be stripped. The lexical
+            case will be preserved.
+
+          ClassName (:term:`string`):
+            Name of the class to be deleted (case independent).
+
+        Raises:
+
+            CIMError: CIM_ERR_NOT_FOUND if ClassName defines class not in
+                CIM repository
+        """
+
+        # Validate namespace and get the class datastore for this namespace
+        class_store = self.get_class_store(namespace)
+        instance_store = self.get_instance_store(namespace)
+
+        if not class_store.exists(ClassName):
+            raise CIMError(
+                CIM_ERR_NOT_FOUND,
+                _format("Class {0!A} in namespace {1!A} not in CIM repository. "
+                        "Nothing deleted.", ClassName, namespace))
+
+        # create list of subclass names and append the target class
+        classnames = self._get_subclass_names(ClassName, class_store, True)
+        classnames.append(ClassName)
+
+        # Delete all instances in this class and subclasses and delete
+        # this class and subclasses
+        for clname in classnames:
+            sub_clns = self._get_subclass_list_for_enums(ClassName, namespace,
+                                                         class_store)
+
+            inst_paths = [inst.path for inst in instance_store.iter_values()
+                          if inst.path.classname in sub_clns]
+
+            for ipath in inst_paths:
+                instance_store.delete(ipath)
```

Comment:
Add a TODO that this should go through the DeleteInstance method of the instance provider, so that user-defined providers get called.

Reply:
**DONE** TODO added